### PR TITLE
properties PLUGGED_IN and BATTERY_LEVEL are never null

### PR DESCRIPTION
**Pull request by AlmasB**

### CODE REVIEW

File: modules/battery/src/main/java/com/gluonhq/attach/battery/impl/IOSBatteryService.java [hunk 1]
```diff
@@ -87,12 +87,12 @@ private static void notifyBatteryState(String state) {
         // ios docs: charging -> device is plugged into power and the battery is less than 100% charged
         // or full -> device is plugged into power and the battery is 100% charged
         boolean plugged = state.equals("Charging") || state.equals("Full");
-        if (PLUGGED_IN != null && PLUGGED_IN.get() != plugged) {
+        if (PLUGGED_IN.get() != plugged) {
```
Comment (tiainen) [inline modules/battery/src/main/java/com/gluonhq/attach/battery/impl/IOSBatteryService.java:90]: Theoretically, they can be `null` if the static initializer block triggers a call to this code (via `initBattery()`), before these two static fields are initialized. Simply moving the initialization of the two fields, above the static initializer block would make this PR valid.

Comment (AlmasB) [reply modules/battery/src/main/java/com/gluonhq/attach/battery/impl/IOSBatteryService.java:90]: Thanks. Yes, this sequence of events had crossed my mind, but after considering `initBattery()` in `Battery.m`, I was reasonably confident that no callbacks are invoked:

```
    mat_jBatteryServiceClass = (*env)->NewGlobalRef(env, (*env)->FindClass(env, "com/gluonhq/attach/battery/impl/IOSBatteryService"));
    mat_jBatteryService_notifyBatteryLevel = (*env)->GetStaticMethodID(env, mat_jBatteryServiceClass, "notifyBatteryLevel", "(F)V");
    mat_jBatteryService_notifyBatteryState = (*env)->GetStaticMethodID(env, mat_jBatteryServiceClass, "notifyBatteryState", "(Ljava/lang/String;)V");

    _battery = [[Battery alloc] init];
```

Having said that, I don't speak `.m`, so not 100% sure my assumptions are valid.

Regardless, happy to move the fields above the static block to remove any confusion about the initialization order.

Comment (tiainen) [reply modules/battery/src/main/java/com/gluonhq/attach/battery/impl/IOSBatteryService.java:90]: Moving the initialization to the top would be the best in the long run.

**Comment by tiainen** [comment]
Before we can merge this PR, could you please [sign the CLA](https://cla.gluonhq.com)? Thanks.

**Review by tiainen — APPROVED** [review]

**Comment by AlmasB** [comment]
Done.

**Pull request merged**